src/openms/source/FORMAT/PercolatorInfile.cpp

```diff
@@ -0,0 +1,335 @@
+        {
+          hit.setMetaValue("charge" + String(i), charge == i);
+        }
+
+        // just first peptide evidence
+        char aa_before = hit.getPeptideEvidences().front().getAABefore();
+        char aa_after = hit.getPeptideEvidences().front().getAAAfter();
+
+        bool enzN = isEnz_(aa_before, unmodified_sequence.prefix(1)[0], enz);
+        hit.setMetaValue("enzN", enzN);
+        bool enzC = isEnz_(unmodified_sequence.suffix(1)[0], aa_after, enz);
+        hit.setMetaValue("enzC", enzC);
+        int enzInt = countEnzymatic_(unmodified_sequence, enz);
+        hit.setMetaValue("enzInt", enzInt);
+
+        hit.setMetaValue("dm", delta_mass);
+        
+        double abs_delta_mass = abs(delta_mass);
+        hit.setMetaValue("absdm", abs_delta_mass);
+        
+        //peptide
+        String sequence = "";
+
+        aa_before = aa_before == '[' ? '-' : aa_before;
+        aa_after = aa_after == ']' ? '-' : aa_after;
+
+        sequence += aa_before;
+        sequence += "."; 
+        // Percolator uses square brackets to indicate PTMs
+        sequence += hit.getSequence().toBracketString(false, true);
+        sequence += "."; 
+        sequence += aa_after;
+        
+        hit.setMetaValue("Peptide", sequence);
+        
+        //proteinId1
+        StringList proteins;
+        for (const PeptideEvidence& pep : hit.getPeptideEvidences())
+        {
+          proteins.push_back(pep.getProteinAccession());
+        }
+        hit.setMetaValue("Proteins", ListUtils::concatenate(proteins, '\t'));
+        
+        StringList feats;
+        for (const String& feat : feature_set)
+        {
+        // Some Hits have no NumMatchedMainIons, and MeanError, etc. values. Have to ignore them!
+          if (hit.metaValueExists(feat))
+          {
+            feats.push_back(hit.getMetaValue(feat).toString());
+          }
+        }
+        if (feats.size() == feature_set.size())
+        { // only if all feats were present add
+          txt.addLine(ListUtils::concatenate(feats, '\t'));
+        }
+      }
+    }
+    return txt;
+  }
+
+  Int PercolatorInfile::getScanNumber_(const String& scan_identifier)
+  {
+    Int scan_number = 0;
+    StringList fields = ListUtils::create<String>(scan_identifier);
+    for (const String& str : fields)
+    {
+      // if scan number is not available, use the scan index
+      Size idx = 0;
+      if ((idx = str.find("scan=")) != string::npos)
+      {
+        scan_number = str.substr(idx + 5).toInt();
+        break;
+      }
+      else if ((idx = str.find("index=")) != string::npos)
+      {
+        scan_number = str.substr(idx + 6).toInt();
+        break;
+      } 
+      else if ((idx = str.find("spectrum=")) != string::npos)
+      {
+        scan_number = str.substr(idx + 9).toInt();
+        break;
+      }
+    }
+    return scan_number;
+  }
+
+  bool PercolatorInfile::isEnz_(const char& n, const char& c, const std::string& enz)
```

Comment:
Potentially finally replace with our Digestion/tokenizer functions.
This would finally also unify the parameter in PercolatorAdapter.

Reply:
I will add a hacktoberfest issue on that